h2o-algos/src/main/java/hex/glm/ConstrainedGLMUtils.java

```diff
@@ -0,0 +1,861 @@
+        } else {
+          constraintCond.add(constrainStr+" = " + oneC._constraintsVal + " and does not satisfy the" +
+                  " condition <= 0");
+          cSatisfied.add("false");
+          allSatisfied = false;
+        }
+        cConditions.add("<= 0");
+      }
+      cValues.add(oneC._constraintsVal);
+    }
+    return allSatisfied;
+  }
+
+  public static String constraint2Str(LinearConstraints oneConst, boolean equalConstr, String startStr, ComputationState state) {
+      boolean isBetaConstraint = oneConst._constraints.size() < 3;
+      StringBuilder sb = new StringBuilder();
+      sb.append(startStr);
+      DataInfo dinfo = state.activeData();
+      boolean standardize = dinfo._normMul != null;
+      List<String> trainNames = stream(dinfo.coefNames()).collect(Collectors.toList());
+      double constantVal = 0;
+      int colInd = -1;
+      int coefOffset = (dinfo._catOffsets == null || dinfo._catOffsets.length == 0) ? 0 : dinfo._catOffsets[dinfo._catOffsets.length - 1];
+      for (String coefName : oneConst._constraints.keySet()) {
+        double constrVal = oneConst._constraints.get(coefName);
+        if (constrVal != 0) {
+          if ("constant".equals(coefName)) {
+            constantVal = constrVal;
+          } else if (trainNames.contains(coefName)) {
+            colInd = trainNames.indexOf(coefName) - coefOffset;
+            if (standardize && colInd >= 0 && !isBetaConstraint) {
+              if (constrVal > 0)
+                sb.append('+');
+              sb.append(constrVal / dinfo._normMul[colInd]);
+            } else {
+              sb.append(constrVal);
+            }
+            sb.append('*');
+            sb.append(coefName);
+          }
+        }
+      }
+      // add constant value here
+      // add constant value to the end
+      if (constantVal != 0) {
+        if (constantVal > 0)
+          sb.append("+");
+        if (isBetaConstraint && colInd >= 0)
+          sb.append(constantVal * dinfo._normMul[colInd]);
+        else
+          sb.append(constantVal);
+      }
+      return sb.toString();
+  }
+  
+  public static List<String> extractConstraintCoeffs(ComputationState state) {
+    List<String> tConstraintCoeffName = new ArrayList<>();
+    boolean nonZeroConstant = false;
+    if (state._equalityConstraintsBeta != null)
+      nonZeroConstant = nonZeroConstant | extractCoeffNames(tConstraintCoeffName, state._equalityConstraintsBeta);
+
+    if (state._lessThanEqualToConstraintsBeta != null)
+      nonZeroConstant = nonZeroConstant | extractCoeffNames(tConstraintCoeffName, state._lessThanEqualToConstraintsBeta);
+
+    if (state._equalityConstraintsLinear != null)
+      nonZeroConstant = nonZeroConstant | extractCoeffNames(tConstraintCoeffName, state._equalityConstraintsLinear);
+
+    if (state._lessThanEqualToConstraints != null)
+      nonZeroConstant = nonZeroConstant | extractCoeffNames(tConstraintCoeffName, state._lessThanEqualToConstraints);
+    
```

Comment:
I would prefer to use the logical or (`||`) instead of the bitwise or (`|`). I think it's correct but I find it confusing and it forces me to think if there can be some potential issues.

Reply:
Done.